src/extensions/content_downloaders/curl_downloader/curl_content_downloader.EXPORTS.cpp

```diff
@@ -37,6 +39,26 @@ ADUC_Result Initialize(const char* initializeData)
     return { ADUC_GeneralResult_Success };
 }
 
+/**
+ * @brief Called on the worker context when execution is beginning.
+ *
+ */
+ADUC_Result OnDownloadBegin()
```

Comment:
my preference (for now) would be an extension initialize/uninitialize.  

adding ondownloadbegin/end without ANY context is useless to an extension author, and appears only serving the purpose of allowing you to call logging init/uninit.